packages/varnish/src/components/VarnishApp.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { ThemeProvider } from '@mui/material/styles';
+import { ThemeProvider as StyledThemeProvider } from 'styled-components';
+
+import { getTheme } from '../theme';
+import { DefaultAppLayoutProvider, LayoutVariant } from './VarnishContext';
+import { ErrorBoundary } from './ErrorBoundary';
+
+export interface VarnishAppProps {
+  layout?: LayoutVariant;
+  theme?: {};
+  children: React.ReactNode | React.ReactNodeArray;
+}
+
+/**
+ * A top level container for your application that handles errors cleanly,
+ * makes the required theme available to child components, and sets up
+ * the application layout.
+ *
+ * If the theme property is provided, it's deeply merged with the default Varnish
+ * theme. Which means you can only provide the properties you'd like to override,
+ * and those not provided will use the defaults.
+ */
+export function VarnishApp({ layout, theme, children }: VarnishAppProps) {
+  const mergedTheme = getTheme('light', theme);
+  return (
+    <StyledThemeProvider theme={mergedTheme}>
```

Comment:
why do we need 2 layers of theme providers?

Reply:
there are actually like 4 :-)
styled components, varnish, material, and materialStyles(i forget them name)